### PR TITLE
ARROW-13280: [R] Bindings for log and trig functions

### PR DESCRIPTION
**Pull request by thisisnic**

### CODE REVIEW

**Comment by thisisnic** [comment]
This PR will need rebasing off https://github.com/apache/arrow/pull/10686 when it's merged

**Comment by lidavidm** [comment]
Sorry for the delay, I've just merged #10686

**Comment by thisisnic** [comment]
> Sorry for the delay, I've just merged #10686

All good, wasn't urgent! :)

File: r/R/expression.R [hunk 1]
```diff
@@ -43,7 +43,17 @@
   "yday" = "day_of_year",
   "hour" = "hour",
   # second is defined in dplyr-functions.R
-  "minute" = "minute"
+  "minute" = "minute",
+  "log" = "ln_checked",
```
Comment (nealrichardson) [inline r/R/expression.R:47]: Now that these exist, we should define the `Math` group generic for Arrow objects, like we have for `Ops`. I made https://issues.apache.org/jira/browse/ARROW-13337 for that.

Comment (nealrichardson) [reply r/R/expression.R:47]: Hmm, in R, `log` is not unary: 

```
log <- function (x, base = exp(1))
```

So we should handle that correctly. If arrow doesn't support general `log_baseN`, (1) make sure there's a JIRA for that, and (2) map the cases that do exist (e, 10, 2, etc.) and for now error "not supported" for any other value of base.

**Review by nealrichardson — CHANGES_REQUESTED** [review]
Looks good, just one request.

File: r/R/expression.R (hunk 1 above)
Comment (thisisnic) [reply r/R/expression.R]: Done and opened https://issues.apache.org/jira/browse/ARROW-13345

File: r/R/dplyr-functions.R [hunk 2]
```diff
@@ -559,3 +559,21 @@ nse_funcs$wday <- function(x, label = FALSE, abbr = TRUE, week_start = getOption
...
+  }
+  
+  if (base == 2) {
+    return(Expression$create("log2_checked", x))
+  }
+  
+  if (base == 10) {
+    return(Expression$create("log10_checked", x))
+  } 
+  stop("`base` values other than exp(1), 2 and 10 not supported in Arrow")
```
Comment (nealrichardson) [inline r/R/dplyr-functions.R:576]:
```suggestion
  # ARROW-13345
  stop("`base` values other than exp(1), 2 and 10 not supported in Arrow")
```

**Review by nealrichardson — APPROVED** [review]

**Pull request closed without merging**